// electron-dialog/dist/index.js
"use strict";
/* IMPORT */
var electron = require("electron");
var isRenderer = (process && process.type === 'renderer'), target = isRenderer ? electron.remote : electron;
/* ELECTRON DIALOG */
var Dialog = {
    options: {
        type: 'none',
        noLink: true
    },
    open: function (options, browserWindow) {
        if (!browserWindow && !isRenderer)
            throw new Error('You need to provide a BrowserWindow object from the main process');
        options = Object.assign({}, Dialog.options, options);
        return target.dialog.showMessageBoxSync(browserWindow || electron.remote.getCurrentWindow(), options);
    },
    alert: function (message, browserWindow) {
        Dialog.open({ message: message }, browserWindow);
    },
    confirm: function (message, browserWindow) {
        return !!Dialog.choice(message, ['Cancel', 'Yes'], browserWindow);
    },
    choice: function (message, buttons, browserWindow) {
        buttons = buttons.slice().reverse();
        var buttonId = Dialog.open({ message: message, buttons: buttons, defaultId: 0 }, browserWindow);
        return buttons.length - buttonId - 1;
    }
};
/* EXPORT */
module.exports = Dialog;
module.exports.default = Dialog;
Object.defineProperty(module.exports, "__esModule", { value: true });
